File: Low Level Design (LLD)/Problems/ParkingLot/Java/model/ParkingFloor.java
package model;

import java.util.*;

public class ParkingFloor {
    private final int floorNumber;
    private final List<ParkingSlot> slots;

    public ParkingFloor(int floorNumber) {
        this.floorNumber = floorNumber;
        this.slots = new ArrayList<>();
    }

    public void addSlot(ParkingSlot slot) {
        slots.add(slot);
    }

    public ParkingSlot getFreeSlot(Vehicle vehicle) {
        for (ParkingSlot slot : slots) {
            if (slot.canFitVehicle(vehicle)) {
                return slot;
            }
        }
        return null;
    }

    public int getFloorNumber() {
        return floorNumber;
    }
}

File: Low Level Design (LLD)/Problems/02. ParkingLot/Cpp/model/src/ParkingFloor.cpp
#include "../include/ParkingFloor.h"

ParkingFloor::ParkingFloor(int floorNumber) : floorNumber(floorNumber) {}

void ParkingFloor::addSlot(ParkingSlot* slot) {
    slots.push_back(slot);
}

ParkingSlot* ParkingFloor::getFreeSlot(Vehicle* vehicle) {
    for (auto& slot : slots) {
        if (slot->canFitVehicle(vehicle)) {
            return slot;
        }
    }
    return nullptr;
}

int ParkingFloor::getFloorNumber() const {
    return floorNumber;
}